Multisig/Presentation/UI Library/QRScanner/QRCodeScannerViewController.swift

```diff
@@ -0,0 +1,158 @@
+        super.viewWillDisappear(animated)
+
+        if (captureSession?.isRunning == true) {
+            captureSession.stopRunning()
+        }
+    }
+    
+    @IBAction func closeButtonTouched(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    func setup() {
+        AVCaptureDevice.requestAccess(for: .video) { [weak self] (result)  in
+            if result {
+                DispatchQueue.main.async {
+                    self?.createScanner()
+                }
+            }
+            else {
+                DispatchQueue.main.async {
+                    self?.presentCameraAccessRequiredAlert()
+                }
+            }
+        }
+    }
+
+    func createScanner() {
+        captureSession = AVCaptureSession()
+
+        guard let videoCaptureDevice = AVCaptureDevice.default(for: .video) else { return }
+        let videoInput: AVCaptureDeviceInput
+
+        do {
+            videoInput = try AVCaptureDeviceInput(device: videoCaptureDevice)
+        } catch {
+            return presentFaildToCreateScannerAlert()
+        }
+
+        if (captureSession.canAddInput(videoInput)) {
+            captureSession.addInput(videoInput)
+        } else {
+            presentFaildToCreateScannerAlert()
+            return
+        }
+
+        let metadataOutput = AVCaptureMetadataOutput()
+
+        if (captureSession.canAddOutput(metadataOutput)) {
+            captureSession.addOutput(metadataOutput)
+            metadataOutput.setMetadataObjectsDelegate(self, queue: DispatchQueue.main)
+            metadataOutput.metadataObjectTypes = [.qr]
+        } else {
+            presentFaildToCreateScannerAlert()
+            return
+        }
+
+        previewLayer = AVCaptureVideoPreviewLayer(session: captureSession)
+        previewLayer.frame = view.layer.bounds
+        previewLayer.videoGravity = .resizeAspectFill
+        view.layer.insertSublayer(previewLayer, at: 0)
+
+        captureSession.startRunning()
+    }
+    
+    private func presentCameraAccessRequiredAlert() {
+        let settingsAppURL = URL(string: UIApplication.openSettingsURLString)!
+        let alert = UIAlertController(
+            title: Strings.cameraAlertTitle,
+            message: Strings.cameraAlertMessage,
+            preferredStyle: .alert
+        )
+        alert.addAction(UIAlertAction(title: Strings.cameraAlertAllow, style: .cancel) { _ in
+            UIApplication.shared.open(settingsAppURL, options: [:], completionHandler: nil)
+        })
+        
+        alert.addAction(UIAlertAction(title: Strings.cameraAlertCancel, style: .default) { [weak self] _ in
+            self?.dismiss(animated: true, completion: nil)
+        })
+        present(alert, animated: true)
+    }
+
+    func presentFaildToCreateScannerAlert() {
+        let ac = UIAlertController(title: Strings.scannerNotSupportedTitle, message: Strings.scannerNotSupportedMessage, preferredStyle: .alert)
```

Comment:
This line is too long and needs to be split:

```suggestion
        let ac = UIAlertController(title: Strings.scannerNotSupportedTitle,
                                   message: Strings.scannerNotSupportedMessage,
                                   preferredStyle: .alert)
```